pkg/api/validation/cluster_validation.go

```diff
@@ -86,6 +88,50 @@ func ValidateScyllaClusterSpec(spec *scyllav1.ScyllaClusterSpec, fldPath *field.
 		}
 	}
 
+	if spec.IngressOptions != nil {
+		allErrs = append(allErrs, ValidateIngressOptions(spec.IngressOptions, fldPath.Child("ingressOptions"))...)
+	}
+
+	return allErrs
+}
+
+func ValidateIngressOptions(options *scyllav1.IngressOptions, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(options.Domain) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("domain"), "domain is required"))
+	} else {
+		for _, msg := range apimachineryvalidation.NameIsDNSSubdomain(options.Domain, false) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("domain"), options.Domain, msg))
+		}
+	}
+
+	if len(options.IngressClassName) != 0 {
+		for _, msg := range apimachineryvalidation.NameIsDNSSubdomain(options.IngressClassName, false) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("ingressClassName"), options.IngressClassName, msg))
+		}
+	}
+
+	if len(options.Annotations) != 0 {
+		allErrs = append(allErrs, apimachineryvalidation.ValidateAnnotations(options.Annotations, fldPath.Child("annotations"))...)
+	}
+
+	hasPortName := len(options.ServiceBackendPort.Name) > 0
+	hasPortNumber := options.ServiceBackendPort.Number != 0
+	if hasPortName && hasPortNumber {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("serviceBackendPort"), "", "cannot set both port name & port number"))
```

Comment:
```suggestion
		allErrs = append(allErrs, field.Forbidden(fldPath.Child("serviceBackendPort"), "cannot set both port name & port number"))
```

Reply:
Fair. Forbidden would make sense in regard to either of the inner fields, for `serviceBackendPort` invalid seems right